spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -259,17 +259,19 @@ public SparkContext createSparkContext() {
 
     for (Object k : intpProperty.keySet()) {
       String key = (String) k;
-      Object value = intpProperty.get(key);
-      logger.debug(String.format("SparkConf: key = [%s], value = [%s]", key, value));
-      conf.set(key, (String) value);
+      String val = toString(intpProperty.get(key));
+      if (!key.startsWith("spark.") || !val.trim().isEmpty()) {
```

Comment:
Then ```val.trim().isEmpty()``` is true, so it will be set for any property, that does not start with "spark." (i.e whitespace as a password)